src/main/java/chessmaster/ChessMaster.java

```diff
@@ -34,8 +40,12 @@ private ChessMaster() {
             difficulty = storage.loadDifficulty();
             currentTurnColor = storage.loadCurrentColor();
             ChessTile[][] existingBoard = storage.loadBoard();
+            board = new ChessBoard(playerColor);
```

Comment:
Should create the a new board to compare to in the `executeSavedMoves` function and let `existingBoard` (from storage) to be used for loading previous game